src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices
+{
+    public static class MarshalTests
+    {
+        [Fact]
+        public static void AllocHGlobal_Int32_Success()
+        {
+            IntPtr p = Marshal.AllocHGlobal(100);
+            Assert.NotEqual(IntPtr.Zero, p);
```

Comment:
In next only writeable is checked. In L#45 you read from that is reallocated, so technically it's not the same. Maybe it is very strict, but if you hypothetically delete your reallocation test, then it Alloc method will be untested.